pkg/losertree/tree.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+// Original version copyright Bryan Boreham, 2024.
+// https://github.com/bboreham/go-loser/tree/any.
+// Loser tree, from https://en.wikipedia.org/wiki/K-way_merge_algorithm#Tournament_Tree
```

Comment:
I am not very familiar with the actual implementation in this library. May I know what was changed and why we need to copy the code here?

Reply:
Nvm you mentioned it in the description...